src/cargopacket.h

```diff
@@ -629,9 +619,9 @@ class StationCargoList : public CargoList<StationCargoList, StationCargoPacketMa
 	{
 		return cp1->source_xy == cp2->source_xy &&
 				cp1->periods_in_transit == cp2->periods_in_transit &&
-				cp1->source_type == cp2->source_type &&
+				cp1->source.type == cp2->source.type &&
 				cp1->first_station == cp2->first_station &&
-				cp1->source_id == cp2->source_id;
+				cp1->source.id == cp2->source.id;
```

Comment:
Could this be `cp1->source == cp2->source`?

Reply:
Certainly.